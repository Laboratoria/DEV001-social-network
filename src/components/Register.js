import { createUser } from '../lib/Auth.js';

export const Register = (onNavigate) => {
  const homeDiv = document.createElement('div');
  homeDiv.className = 'contenedor';

  const textoRegister = document.createElement('h2');
  textoRegister.textContent = 'Puedes registrarte en Semillero usando tu correo electrónico. Tu clave debe tener un mínimo de seis dígitos:';

  const registerMail = document.createElement('input');
  registerMail.placeholder = 'ingresa tu correo';
  registerMail.type = 'email';

  const registerPass = document.createElement('input');
  registerPass.placeholder = 'crea una contraseña';
  registerPass.type = 'password';

  const errorSpace = document.createElement('p');
  errorSpace.id = 'errorSpace';

  const buttonSend = document.createElement('button');
  buttonSend.textContent = 'CREAR USUARIO';
  buttonSend.classList.add('register');

  const buttonHome = document.createElement('button');
  buttonHome.textContent = 'VOLVER AL INICIO';
  buttonHome.addEventListener('click', () => {
    onNavigate('/');
  });

  buttonSend.addEventListener('click', () => {
    const userMail = registerMail.value;
    const userPass = registerPass.value;
    createUser(userMail, userPass, onNavigate);
  });

  homeDiv.append(textoRegister, registerMail, registerPass, errorSpace, buttonSend, buttonHome);
  return homeDiv;
};
