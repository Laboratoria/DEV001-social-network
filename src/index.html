<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="stylesheet" href="./lib/archivo_css/styles.css" >
  <link rel="stylesheet" href="./lib/archivo_css/welcome.css">
  <link rel="stylesheet" href="./lib/archivo_css/signin.css">
  <link rel="stylesheet" href="./lib/archivo_css/signup.css">
  <link rel="stylesheet" href="./lib/archivo_css/feed.css">
  <!-- <link rel="stylesheet" href="feed.css"> -->
  <title>FestiFan</title>
</head>
<body>
  <!--  En este nodo se carga todo -->
  <div id="root"></div>
  <script type="module" src="main.js"></script>
</body>
</html>