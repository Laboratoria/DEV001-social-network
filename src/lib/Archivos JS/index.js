// aqui exportaras las funciones que necesites
import {
  updateProfile,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  GoogleAuthProvider,
  signInWithPopup,
} from 'firebase/auth';
import { auth } from './firebase.js';

export const functionRegister = async (email, password, name) => {
  //console.log(email);
  //console.log(password);
  //console.log(name);

  try {
    const userCredentials = await createUserWithEmailAndPassword(
      auth,
      email,
      password
    );
    await updateProfile(userCredentials.user, {
      displayName: name,
    });
    //console.log(userCredentials.user);
    // console.log(userCredentials.user.displayName = name);
    return userCredentials.user;
  } catch (error) {
    //console.log(error);

    if (error.code === 'auth/email-already-in-use') {
      alert('El correo ya está registrado');
      return 'error';
    }
    if (error.code === 'auth/invalid-email') {
      alert('Debes ingresar un correo válido');
      return 'error';
    }
    if (error.code === 'auth/weak-password') {
      alert('La contraseña debe tener al menos 6 carácteres');
      return 'error';
    }
    if (error.code) {
      alert('Algo está mal en tu registro');
      return 'error';
    }
  }
};

export const functionLogin = async (email, password) => {
  //console.log(email);
  //console.log(password);
  try {
    const credentials = await signInWithEmailAndPassword(auth, email, password);
    //console.log(credentials.user);
    return credentials.user;
  } catch (error) {
    if (error.code === 'auth/wrong-password') {
      alert('Contraseña incorrecta');
      return 'error';
    }
    if (error.code === 'auth/user-not-found') {
      alert('Dirección Email no encontrada, por favor regístrese');
      return 'error';
    }
    if (error.code) {
      alert('Error en inicio de sesión, intente nuevamente');
      return 'error';
    }
  }
};

export const functionRegisterGoogle = async () => {
  const provider = new GoogleAuthProvider();
  try {
    const credentials = await signInWithPopup(auth, provider);
    //console.log(credentials.user);
    return credentials;
  } catch (error) {
    //console.log(error);
    return 'error';
  }
};
