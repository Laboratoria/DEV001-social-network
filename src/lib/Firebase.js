// Import the functions you need from the SDKs you need
import { initializeApp } from './Imports.js';
// import { getFirestore } from 'firebase/firestore';

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
  apiKey: 'AIzaSyBhg4DSAWYv5vj5PvFXKIXRuZSP-O3HtNo',
  authDomain: 'sn-semillero.firebaseapp.com',
  projectId: 'sn-semillero',
  storageBucket: 'sn-semillero.appspot.com',
  messagingSenderId: '231094255194',
  appId: '1:231094255194:web:9f8e081918de3fd7436263',
  measurementId: 'G-0KGE235LCY',
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);

// Initialize Cloud Firestore and get a reference to the service
// const db = getFirestore(app);
