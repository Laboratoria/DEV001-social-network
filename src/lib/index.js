// aqui exportaras las funciones que necesites
// import { onNavigate } from '../main.js'