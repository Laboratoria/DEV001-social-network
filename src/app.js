import { Home } from './componentes/Home';
import { Register } from './componentes/Register';
import { goTo } from './App/routes';
import { NotFound } from './componentes/404';
import { Wall } from './componentes/Wall';
import { Post } from './componentes/Post';

const routes = {
  '/': Home,
  '/register': Register,
  '/404': NotFound,
  '/post': Post,
  '/wall': Wall,
};

export const rootDiv = document.getElementById('root');

export const changeHash = (hash) => {
  window.location.hash = hash;
};
const path = window.location.pathname;
goTo(path);

window.addEventListener('hashchange', () => {
  let path = window.location.pathname;
  if (routes[path] === undefined) {
    path = '/404';
    path = '/404';
    goTo(path);
  } else {
    routes[path](rootDiv, changeHash);
  }
}, false);
