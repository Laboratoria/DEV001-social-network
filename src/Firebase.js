 // Import the functions you need from the SDKs you need
 import { initializeApp } from "firebase/app";
 import { getAnalytics } from "firebase/analytics";
 import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, GoogleAuthProvider, signInWithPopup } from "firebase/auth";
 import { onSnapshot, query, getFirestore, collection, addDoc } from "firebase/firestore";
 // TODO: Add SDKs for Firebase products that you want to use
 // https://firebase.google.com/docs/web/setup#available-libraries

 // Your web app's Firebase configuration
 // For Firebase JS SDK v7.20.0 and later, measurementId is optional
 const firebaseConfig = {
     apiKey: "AIzaSyDHhWzXY4iG1hZ-A7ll2DAHR-qMTprOwxg",
     authDomain: "red-social-49f38.firebaseapp.com",
     projectId: "red-social-49f38",
     storageBucket: "red-social-49f38.appspot.com",
     messagingSenderId: "784877427563",
     appId: "1:784877427563:web:cf8fb0a4b452b4f62a0a2e",
     measurementId: "G-T2EWJK77KP"
 };

 // Initialize Firebase
 const app = initializeApp(firebaseConfig);
 const analytics = getAnalytics(app);
 const auth = getAuth(app);
 const provider = new GoogleAuthProvider();
 const db = getFirestore(app);

 export const subscribeCollection = () => {
     const q = query(collection(db, "Post"));
     const unsubscribe = onSnapshot(q, (querySnapshot) => {
         querySnapshot.forEach((doc) => {
             console.log(doc.data());
         });
     });
 }
 subscribeCollection();

 export const registrarUsuario = (email, contraseña) => { //usamos mail y contraseña porque no están en Firebase
     return createUserWithEmailAndPassword(auth, email, contraseña); //debe contener tal cual el nombre de los parámetros de arriba
 }


 export const loginUsuario = (email, contraseña) => {
     return signInWithEmailAndPassword(auth, email, contraseña);
 }


 export const registroGoogle = () => {
     return signInWithPopup(auth, provider);
 }

 export const publicarPost = (postValue) => {
     const user = auth.currentUser;
     console.log(user);
     const docRef = addDoc(collection(db, "Post"), {
         contenidoPost: postValue,
         nombreUsuario: user.displayName,
         email: user.email,
         date: new Date(),

     });
     return docRef;
 }


 //el displayName sólo funciona con Google


 //  export const obtenerUsuario = ('Post', uid) => {
 //     addDoc(collection(db,'Post'), {
 //         contenidoPost: postValue, uid
 //     })
 //  }


 //  export const obtenerUsuario = (auth, (user) => {
 //      if (user) {
 //          user.email;
 //          // ...
 //      } else {
 //          // User is signed out
 //          // ...
 //      }
 //  });



 //  export const publicarPost = (postValue, idAutor) => {
 //     const docRef = addDoc(collection(db, "Post"), {
 //         contenidoPost: postValue,
 //         idAutor: idAutor,
 //     });
 //     return docRef;
 // }