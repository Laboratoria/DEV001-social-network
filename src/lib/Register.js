// import { getAuth, sendSignInLinkToEmail } from "firebase/auth";
import { registrarUsuario } from "../Firebase";


export const Register = (onNavigate) => {
    const HomeDiv = document.createElement('div');
    HomeDiv.className = 'contenedorRegister';
    const buttonHome = document.createElement('button');
    buttonHome.className = 'salida1';

    buttonHome.textContent = 'Regresar al inicio';

    buttonHome.addEventListener('click', () => onNavigate('/'));
    HomeDiv.appendChild(buttonHome);

    //LOGO
    const logoHome = document.createElement('img');
    logoHome.src = '../Imagenes/Logo.png';
    logoHome.className = 'logoHome';
    HomeDiv.appendChild(logoHome);

    //Frase Bajada
    const frase = document.createElement('p');
    frase.className = 'frase';
    frase.textContent = 'Tu red social especializada en Manga'
    HomeDiv.appendChild(frase);

    //Inicia sesión título
    const frase2 = document.createElement('p');
    frase2.className = 'frase2';
    frase2.textContent = 'Regístrate';
    HomeDiv.appendChild(frase2);


    //INPUT EMAIL
    const inputEmail = document.createElement('input'); // .placeholder = 'Email';
    inputEmail.className = 'correo';
    inputEmail.placeholder = 'Email';
    inputEmail.type = 'email';
    HomeDiv.appendChild(inputEmail);

    //INPUT PASSWORD
    const inputpassword = document.createElement('input');
    inputpassword.className = 'password';
    inputpassword.placeholder = 'Contraseña';
    inputpassword.type = 'password';
    HomeDiv.appendChild(inputpassword);



    //BOTÓN REGISTRARTE / BOTÓN INPUT
    const buttonRegistro = document.createElement('button');
    buttonRegistro.textContent = 'Registrarte';
    buttonRegistro.className = 'botonregistro';
    HomeDiv.appendChild(buttonRegistro);

    buttonRegistro.addEventListener('click', (e) => {
        e.preventDefault()

        let email = document.getElementsByClassName('correo')[0].value;
        let contraseña = document.getElementsByClassName('password')[0].value;

        registrarUsuario(email, contraseña).then((exito) => {
            // obtenerUsuario(user.email);
            console.log(exito);

            onNavigate('/wall');

        }).catch((error) => {
            console.log(error);
            const registerError = document.createElement('p');
            registerError.innerHTML = ''; //NO PUEDO LIMPIAR EL MENSAJE ANTES
            registerError.className = 'error';

            registerError.textContent = 'Algo ha salido mal, inténtelo de nuevo';

            HomeDiv.appendChild(registerError);


            // ...
        });
    })

    //CUADRO TRANSPARENTE
    const cuadroDivRegister = document.createElement('div');
    cuadroDivRegister.className = 'cuadroTransparente2';
    HomeDiv.appendChild(cuadroDivRegister);

    //FONDO
    const fondoRegister = document.createElement('img');
    fondoRegister.src = '../Imagenes/Register.jpg';
    fondoRegister.className = 'fondoRegister';
    HomeDiv.appendChild(fondoRegister);

    return HomeDiv;

};


// const auth = getAuth();
// signInWithPopup(auth, provider)
//     .then((result) => {
//         // This gives you a Google Access Token. You can use it to access the Google API.
//         const credential = GoogleAuthProvider.credentialFromResult(result);
//         const token = credential.accessToken;
//         // The signed-in user info.
//         const user = result.user;
//         // ...
//     }).catch((error) => {
//         // Handle Errors here.
//         const errorCode = error.code;
//         const errorMessage = error.message;
//         // The email of the user's account used.
//         const email = error.customData.email;
//         // The AuthCredential type that was used.
//         const credential = GoogleAuthProvider.credentialFromError(error);
//         // ...
//     });