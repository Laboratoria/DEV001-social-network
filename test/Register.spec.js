/**
 * @jest-environment jsdom
 */
import { registrarUsuario, createUserWithEmailAndPassword, getAuth } from "../src/Firebase";
import { Register } from "../src/lib/Register";

jest.mock('../src/main.js');
jest.mock('../src/Firebase.js');

describe('Los test del Registro', () => {
    test('Comprobar que exista un boton para registrarse', () => {
        const register = Register()
        const boton = register.querySelector('.botonregistro');
        expect(boton).not.toBeNull();


        // boton.click()
        // const error = register.querySelector('.error');
        // expect(error.textContent).toEqual('Algo ha salido mal, inténtelo de nuevo');

    })
    test('Comprobar que exista un input para el email', () => {
        const register = Register()
        const input = register.querySelector('.correo');
        expect(input).not.toBeNull();


    });

    it('Snapshot del registro', () => {
        expect(Register()).toMatchSnapshot();
    });

    test('Deberia retornar un objeto con la propiedad email y password', () => {
        registrarUsuario('persona1@gmail.com', '1234567');
        expect({
            email: 'persona1@gmail.com',
            password: '1234567',
        }).toEqual(expect.anything());
    });

    it('registrarUsuario debería ser una función', () => {
        expect(typeof registrarUsuario).toBe('function');
    });



})


// it('Ejecuta registrarUsuario()', () => {
//     let email = 'persona@gmail.com';
//     let contraseña = 'persona123';
//     const registro = registrarUsuario(email, contraseña);
//     expect(registro).toBeTruthy();
// });

// expect(registrarUsuario).toHaveBeenCalled();


// segundoTest('El usuario al hacer click en el botón de registro sin llenar correo o contraseña aparece un error', () => {
//     const register = Register()
//     const error = register.querySelector('.error');
//     const boton = register.querySelector('.botonregistro');
//     boton.click()
//     const correo = register.querySelector('.correo');
//     const contraseña = register.querySelector('.password');

// document.body.innerHTML = '<input class="correo"></input> <input class="password"></input>';