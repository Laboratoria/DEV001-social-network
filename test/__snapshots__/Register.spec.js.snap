// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Los test del Registro Snapshot del registro 1`] = `
<div
  class="contenedorRegister"
>
  <img
    class="logoHome"
    src="../Imagenes/Logo.png"
  />
  <p
    class="frase"
  >
    Tu red social especializada en Manga
  </p>
  <p
    class="frase2"
  >
    Regístrate
  </p>
  <input
    class="correo"
    placeholder="Email"
    type="email"
  />
  <input
    class="password"
    placeholder="Contraseña"
    type="password"
  />
  <button
    class="botonregistro"
  >
    Registrarte
  </button>
  <button
    class="salida1"
  >
    Regresar al inicio
  </button>
  <div
    class="cuadroTransparente2"
  />
  <img
    class="fondoRegister"
    src="../Imagenes/Register.jpg"
  />
</div>
`;
